### PR TITLE
Enable vault name mapping and error suppression

### PR DESCRIPTION
**Pull request by yash-nisar**
# Description

Enable vault name mapping and error suppression

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/dotnet-sdk/issues/887

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: src/Dapr.Extensions.Configuration/DaprSecretDescriptor.cs [hunk 1]
```diff
@@ -29,22 +29,46 @@ public class DaprSecretDescriptor
...
+        {
+            
+        }
+        
+        /// <summary>
+        /// Secret Descriptor Constructor
         /// </summary>
-        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata)
+        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata, 
+            bool isRequired, string secretKey)
```
Comment (halspang) [inline src/Dapr.Extensions.Configuration/DaprSecretDescriptor.cs:66]: From a style standpoint, if you have to split the parameters, it's best to put each parameter on its own line instead.

File: src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs [hunk 2]
```diff
@@ -197,16 +201,39 @@ private async Task LoadAsync()
...
+                    }
+                    catch (DaprException e)
+                    {
+                        if (secretDescriptor.IsRequired)
+                        {
+                            throw e;
+                        }
+                            
+                        Console.WriteLine($"'{secretDescriptor.SecretName}' doesn't exists in Key Vault but because " +
+                                          $"it doesn't require existence so all errors suppressed!");
```
Comment (halspang) [inline src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs:221]: I would remove this unless there is a logger available. Then you could make it a debug statement.

File: src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs [hunk 3]
```diff
@@ -197,16 +201,39 @@ private async Task LoadAsync()
...
                         {
-                            throw new InvalidOperationException($"A duplicate key '{key}' was found in the secret store '{store}'. Please remove any duplicates from your secret store.");
+                            throw new InvalidOperationException($"A duplicate key '{key}' was found in the " +
+                                                                $"secret store '{store}'. Please remove any " +
+                                                                $"duplicates from your secret store.");
                         }
 
-                        data.Add(normalizeKey ? NormalizeKey(key) : key, result[key]);
+                        data.Add(normalizeKey ? NormalizeKey(secretDescriptor.SecretName) : secretDescriptor.SecretName, 
+                            result[key]);
```
Comment (halspang) [inline src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs:236]: You're doing a lot of line splitting here. What do you have your margin set to? 120-160 characters is a good width for modern screens (I prefer 160).

File: test/Dapr.Extensions.Configuration.Test/DaprSecretStoreConfigurationProviderTest.cs [hunk 4]
```diff
@@ -167,29 +168,61 @@ public void LoadSecrets_FromSecretStoreThatReturnsOneValue()
...
+            var config = CreateBuilder()
+                .AddDaprSecretStore(storeName, secretDescriptors, daprClient.Object)
                 .Build();
 
+            config[firstSecretKey].Should().Be(firstSecretValue);
+            config[secondSecretKey].Should().Be(secondSecretValue);
+        }
+        
+        [Fact]
+        public void LoadSecrets_FromSecretStoreWithADifferentSecretKeyAndName()
```
Comment (halspang) [inline test/Dapr.Extensions.Configuration.Test/DaprSecretStoreConfigurationProviderTest.cs:202]: We should really have a test for `required = false` as well.

**Review by halspang — CHANGES_REQUESTED** [review]
Seems good overall, just a few comments.

File: src/Dapr.Extensions.Configuration/DaprSecretDescriptor.cs (hunk 1 above)
Comment (yash-nisar) [reply src/Dapr.Extensions.Configuration/DaprSecretDescriptor.cs:66]: Fixed

File: src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs (hunk 2 above)
Comment (yash-nisar) [reply src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs:221]: Fixed

File: src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs (hunk 3 above)
Comment (yash-nisar) [reply src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs:236]: Fixed

**Comment by yash-nisar** [comment]
Closing in favour of https://github.com/dapr/dotnet-sdk/issues/887.

**Pull request closed without merging**